game/world/managers/objects/units/creature/utils/TrainerUtils.py

```diff
@@ -148,3 +149,16 @@ def can_train(creature_mgr, player_mgr) -> bool:
 
         # Mount, TradeSkill or Pet trainer.
         return True
+
+    @staticmethod
+    def player_can_learn_magic_talent(training_spell: TrainerTemplate, spell: Spell, player_mgr) -> bool:
+        # Mana users can use resists, and talents for skills their class can use (including wands.)
+        if player_mgr.power_type == PowerTypes.TYPE_MANA:
+            if not player_mgr.skill_manager.has_skill(training_spell.reqskill) and training_spell.reqskill != 0:
+                return False
+        # Non-mana users can only use resist magic talents.
+        else:
+            if 'Resist' not in spell.Name_enUS:
+                return False
+        return True
+
```

Comment:
Checking specifically for magic talents doesn't seem necessary. Since this PR adds skill requirements to talents, we could instead now use the skills' Race/ClassMasks and get the same (or better) result.